Batch calc

ojomio opened the pull request:
Added possibility to calc function call results missing in cache in a "bulk" fashion

This way a function which accepts a vector can efficiently be called with a vector of missing input arguments, effectively enabling "partial" arg caching

taylorhakes commented:
@ojomio Please create a ticket before doing all this work. I like the idea behind this feature, but I am not sure I am sold on the API.  This library is built around single function caching. I think partial parameter caching is kind of a different feature, so I would make a different function instead of overloading @cache.cache for that. Maybe something like `@cache.cache_partial()`. Would you be open to that instead? 

Please don't reformat the code as part of this change. That is a separate thing and I prefer the current styles

ojomio commented:
It is just how we implemented this for the needs of our use at my company. I implemented this and proposed for the general use in the pr, though I am totally open on the exact layout07:48, 24 февраля 2022 г., Taylor Hakes ***@***.***>:
@ojomio Please create a ticket before doing all this work. I like the idea behind this feature, but I am not sure I am sold on the API.  This library is built around single function caching. I think partial parameter caching is kind of a different feature, so I would make a different function instead of overloading @cache.cache for that. Maybe something like @cache.cache_partial(). Would you be open to that instead?
Please don't reformat the code as part of this change. That is a separate thing and I prefer the current styles

—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.
You are receiving this because you were mentioned.Message ID: ***@***.***>
-- Отправлено из мобильного приложения Яндекс Почты

taylorhakes commented:
Closing due to no response. Please reopen if someone wants to add tests
